packages/es-components/src/components/base/icons/Icon.js

```diff
@@ -2,48 +2,36 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
 
-import { regular } from './icon-definitions';
-
 const StyledIcon = styled.i`
   display: inline-block;
-  font-family: bds-func-icons !important;
-  font-size: ${props => props.fontSize};
-  font-style: normal;
-  font-weight: inherit;
-  font-variant: normal;
-  line-height: 1;
-  speak: none;
+  font-size: ${props => props.size};
   text-decoration: none;
-  text-transform: none;
-  -webkit-font-smoothing: antialiased;
-  -moz-osx-font-smoothing: grayscale;
   vertical-align: text-bottom;
-
-  &:before {
-    content: "\\e${props => props.content}";
-  }
 `;
 
-function Icon({ name, size, ...other }) {
-  const iconName = `im-icon-${name}`;
-
-  const styledIconProps = {
-    content: regular[iconName],
-    fontSize: size !== undefined ? `${size}px` : 'inherit'
-  };
-
-  return <StyledIcon aria-hidden {...styledIconProps} {...other} />;
+function Icon({ name, size, className, ...other }) {
+  return (
+    <StyledIcon
+      className={`bds-icon bds-${name}${className ? ` ${className}` : ''}`}
```

Comment:
You should be able to use `${className || ''}` instead of a ternary here.

Reply:
I don't think that will work with the conditional space I want before the `className`, if it exists. (I dislike the empty trailing space in the className value you'd otherwise see.)

Reply:
I was able to make it more readable combining your suggestion with a `trim()`. :)

Reply:
That works! I missed the extra space in there!